### DISCUSSION TITLE
plausible.js:1 POST /api/event 500

### DISCUSSION THREAD

**Discussion by gaetandezeiraud** [plausible/analytics#706] [Self-Hosted Support] [answered]
Hi, 
I have setup a self-hosted version of Plausible. But I have an error (in dev or production). 
`plausible.js:1 POST https://analytics.dezeiraud.com/api/event 500`. 

In docker, I have this error: 
```
Request: POST /api/event

** (exit) an exception was raised:

** (BadMapError) expected a map, got: nil

(elixir 1.11.3) lib/map.ex:469: Map.get(nil, :country, nil)

(plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:162: PlausibleWeb.Api.ExternalController.visitor_country/1

(plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:87: PlausibleWeb.Api.ExternalController.create_event/2

(plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:9: PlausibleWeb.Api.ExternalController.event/2

(plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:1: PlausibleWeb.Api.ExternalController.action/2

(plausible 0.0.1) lib/plausible_web/controllers/api/external_controller.ex:1: PlausibleWeb.Api.ExternalController.phoenix_controller_pipeline/2

(phoenix 1.5.7) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2

(plausible 0.0.1) lib/plug/error_handler.ex:65: PlausibleWeb.Router.call/2
```

Any idea?
Thanks!

**Comment by Vigasaurus** [comment]
Sadly this exact issue has come up before, and all that was able to come up troubleshooting-wise was to ensure the Geolocation service was correctly set up - but as I understand the issue wasn't fully solved for the other asker as well. https://github.com/plausible/analytics/discussions/638

Best bet I'd say is for you to post your docker-compose and the existence of the geoip directory in your docker-compose command, but that older post may just solve your issue.

**Reply by gaetandezeiraud** [reply]
Thank you. Not possible to disable this feature? I have try different things but nothing worked. And my geolite key is used. And I can get the file if I change the volume. So, I don't understand.

And, in my memories, it's possible to use Plausible without this system. But even if a clean setup, I have the error ... Because in reality, I don't really need geolocalisation for the moment.

**Reply by ukutaht** [reply]
I cannot reproduce this error on on the `latest` tag which is equivalent to `v1.2`. Are you using a different tag? If not, could you put together a reproducible case for this? Sorry I'm a bit stuck without being able to repro

**Reply by gaetandezeiraud** [reply]
Hi,
Yes, I have try with differents versions. Same problem. I just set up the project with the è docker-composer-up` command. Following the documentation. And after, because I have the error, setup (with the documentation and the original script) geoip. Without success.

Sorry, but I don't know how I can help you more. I have change nothing. (Just path for Windows). https://github.com/Brouilles/hosting/blob/master/docker-compose.yml

I have decided to stay on GA for the moment. No more time. Sorry.

**Comment by gaetandezeiraud** [comment, marked as answer]
duplicate of https://github.com/plausible/analytics/discussions/638
